test/run_e2e_test.py

```diff
@@ -0,0 +1,195 @@
+                            '/linux_system_memory_collector',
+                            '/listener_process_cpu_node',
+                            '/listener_process_memory_node',
+                            '/talker_process_cpu_node',
+                            '/talker_process_memory_node']
+
+EXPECTED_LIFECYCLE_STATE = 'active [3]'
+LAUNCH_COMMAND = 'ros2 launch system_metrics_collector talker_listener_example.launch.py'
+TIMEOUT_SECONDS = 20
+
+
+class StatisticsListener(Node):
+    """Listen for MetricsMessages published on the /system_metrics topic."""
+
+    def __init__(self, future: Future, expected_lifecycle_nodes: List):
+        super().__init__('statisticsListener')
+
+        self.sub = self.create_subscription(MetricsMessage,
+                                            'system_metrics',
+                                            self.listener_callback,
+                                            1)
+        self.future = future
+        self.received_all_published_stats = False
+        self.expected_lifecycle_nodes = expected_lifecycle_nodes
+
+    def listener_callback(self, msg) -> None:
+        """
+        Handle published MetricsMessages.
+
+        :param msg: received message
+        :return: None
+        """
+        node_name = '/' + msg.measurement_source_name
+
+        if node_name in self.expected_lifecycle_nodes:
+            self.expected_lifecycle_nodes.remove(node_name)
+
+        if not self.expected_lifecycle_nodes:
+            self.received_all_published_stats = True
+            self.future.set_result(self.received_all_published_stats)
+
+
+def print_test(message: str) -> None:
+    """
+    Print a formatted string for easier parsing.
+
+    :param message: string to be printed
+    :return:
+    """
+    print('========================')
+    print(message)
+    print('========================')
+
+
+def check_for_expected_nodes(args=None) -> None:
+    """
+    Check that all expected nodes can be found.
+
+    :param args:
+    :return:
+    """
+    expected_nodes = ['/listener', '/talker'] + EXPECTED_LIFECYCLE_NODES
+
+    rclpy.init(args=args)
+    node = rclpy.create_node('check_for_expected_nodes_test')
+    actual_nodes = node.get_node_names_and_namespaces()
+    node.destroy_node()
+    rclpy.shutdown()
+
+    expected_count = len(expected_nodes)
+    actual_count = 0
+    for item in actual_nodes:
+        found_node = '/' + item[0]
+        if found_node in expected_nodes:
+            actual_count += 1
+            expected_nodes.remove(found_node)
+
+    if expected_count != actual_count:
+        raise Exception('Failed to enumerate expected nodes. Found #' + actual_count + ': ', actual_nodes)
+    else:
+        print_test('check_for_expected_nodes success')
+
+
+def check_lifecycle_node_enumeration() -> None:
+    """
+    Check that all lifecycle nodes exist.
+
+    This requires the ros2lifecycle dependency.
+    :return:
```

Comment:
If return is `None` do we still need it in the docstrings?

Reply:
The [Google Style guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md) says it's optional, so I'll remove.